src/rulesets/LinearAlgebra/matfun.jl

```diff
@@ -0,0 +1,243 @@
+# matrix functions of dense matrices
+
+# NOTE: for matrix functions whose power series representation has real coefficients,
+# the pullback and pushforward are related by an adjoint.
+# Specifically, if the pushforward of f(A) is (f_*)_A(ΔA), then the pullback at Y=f(A) is
+# (f^*)_Y(ΔY) = (f_*)_{A'}(ΔY) = ((f_*)_A(ΔY'))'
```

Comment:
Well that is hideous, but notation is hard, and harder in unicode.

Reply:
>What if we just made a section for this in the docs, (maybe as internal notes or something)
> and wrote the latex and then linked to that?

Hm, that's an idea. I'll consider it, potentially for a future PR.

> This bit makes sense:
> `(f^*)_Y(ΔY) = ((f_*)_A(ΔY'))'`
> so the pullback at `A`, i.e. the pullback from Y...is equal to the the adjoint of the pushing forward at `A`, the adjoint of of the output senstivity.

Ah yes your description is correct (although it's the adjoint of the pushing forward of the adjoint). I just checked Lee, and this should be the right notation:
```suggestion
# (f^*)_A(ΔY) = (f_*)_{A'}(ΔY) = ((f_*)_A(ΔY'))'
```

> (though that's not well defined since not all functions are monotonic?)

I'm not sure what you mean by this.

> Magical expodential symmetry? (I feel like i made the same suprised sounds for the same reason on your last PR)

It's still surprising to me. Although this property is general for all of the matrix functions defined in LinearAlgebra, not just `exp`. It doesn't follow for all matrix functions though, just those whose convergent power series have *real* coefficients.